lib/appnexusapi/faraday/raise_http_error.rb

```diff
@@ -25,11 +29,21 @@ def on_complete(response)
           when 503
             raise AppnexusApi::ServiceUnavailable, error_message(response)
           end
+
+          return if response.body.empty?
+          body = JSON.parse(response.body).fetch('response', {})
+
+          if body['error_code'] == RATE_EXCEEDED_ERROR
+            raise AppnexusApi::RateLimitExceeded, "Retry after #{response.response_headers['retry-after']}s"
+          elsif body['error_id'] && !body['error_id'].empty?
+            # TODO: this should raise an error, but a lot of the specs currently have error responses
```

Comment:
this is kind of a bummer; not sure this is because of my account's access and @infinite-monkeys will be able to replace the `.yml` files but the specs were passing even with these error responses (in fact this kind of error response never even generated any errors, which is ... problematic)